source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,23 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
```

Comment:
I'm confused but `std::string::compare` is what I was using. So should I use pointer walk through or `string::compare`?